### ISSUE TITLE
[2020 ICLR Oral] Your Classifier is Secretly an Energy-Based Model and You Should Treat it Like One

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#91]
원래 generative model은 semi-supervised learning, imputation of missing data, calibration of uncertainty와 같은 downstream problem에 대해 장점이 있음

하지만 근래의 strongest generative modeling approach와 state-of-the-art discriminative model간의 performance gap이 큼

해당 논문은 energy based model (EBM)의 관점으로 discriminative model을 해석하여, generative 또한 가능한 hybrid model을 만들어 discriminative와 generative 두 영역에서 좋은 성능을 냄

논문의 3가지 main contribution은 다음과 같음

1. 모든 discriminative model에 generative model이 숨겨져있음을 보이고, 이를 이용하여 Joint Energy-based Model (JEM)을 제시
2. Generative와 discriminative을 동시에 학습하는 다른 hybrid model의 성능을 outperform
3. Generative modeling의 incorporation으로 인해 calibration, out-of-distribution detection, adversarial robustness에 improvement

Limitation으로는 학습이 불안정하고 hyperparameter에 sensitive한 것

논문의 핵심인 JEM에 대해서만 설명

## 1. Energy Based Models
<img width="450" alt="image" src="https://user-images.githubusercontent.com/49643709/194348887-d724e976-0b06-4a29-b585-c34ad79a6b70.png">

여기서 분모인 partition function이 intractable함
(estimating normalized densities is intractable and standard maximum likelihood estimation of the parameters $\theta$ is not straightforward)

학습을 하기 위해 single example $\textrm{x}$에 대해 log-likelihood의 $\theta$에 대한 derivative는 다음과 같음

<img width="900" alt="image" src="https://user-images.githubusercontent.com/49643709/194349884-31a9cc1a-9f10-4ccc-8a0f-888cc3ff3b48.png">

여기서 expectation 항을 보면, $p_{\theta}(\textrm{x})$에서 sample을 하는게 어려움

이를 Stochastic Gradient Langevin Dynamics (SGLD)를 이용하여 해결

<img width="900" alt="image" src="https://user-images.githubusercontent.com/49643709/194350330-c546cd0f-7950-403b-b60f-c472e0c07f9e.png">

즉 간단히 이야기하면 EBM이 학습하기 어려운 요소가 partition function이 intractable하여 log-likelihood의 derivative를 구했고, 이를 또 구해보니 sampling을 하는게 어려워 SGLD를 통해 해결했다는 것임

Log-likelihood의 derivative 관련 유도는 comment에 달아두었으니 참고

## 2. What Your Classifier is Hiding

Discriminative model의 경우 다음과 같이 categorical distribution을 구할 수 있음

<img width="900" alt="image" src="https://user-images.githubusercontent.com/49643709/194358121-153a049a-5506-4a3e-aed5-7e6c06226b2c.png">

$p_{\theta}(y|x)$를 bayes rule을 이용해서 나눠보면 $p_{\theta}(x, y)$와 $p_{\theta}(x)$를 다음과 같이 표현할 수 있음

<img width="900" alt="image" src="https://user-images.githubusercontent.com/49643709/194360067-22f1ca62-7e22-4db6-86f5-11de8b91b9b3.png">

위 식에서의 $Z(\theta)$는 unknown normalized constant이고, 종합하면 다음과 같음

<img width="350" alt="image" src="https://user-images.githubusercontent.com/49643709/194344167-157f394c-dfd1-4ad2-8d23-02250987a298.png">

## 3. Optimization

총 3가지 loss가 있으며, 이를 목적에 맞게 종합하여 사용

> 1. $\mathrm{log} p(y|x)$ : cross entropy loss
> 2. $\mathrm{log} p(x)$ : data의 energy와 sample의 energy 차이
>     <img width="600" alt="image" src="https://user-images.githubusercontent.com/49643709/194369796-c40d652c-e61f-4896-8063-14bcdd9cdc97.png">
> 3. $\mathrm{log} p(x, y)$ : 1번과 2번 합으로 계산 (unconditional sampling이 아니라 class conditional sampling)
>     <img width="600" alt="image" src="https://user-images.githubusercontent.com/49643709/194369754-2d6ba9ed-624f-4be1-9c56-054ad820bca6.png">

- Class conditional JEM : 1번, 2번 3번 loss 합해서 사용
- Unconditional JEM : 1번, 2번 loss 합해서 사용

Unconditional JEM이 classification 성능은 더 좋고, class conditional JEM은 generation을 더 잘한다고 함

논문에서는 unconditional JEM을 사용하여 실험 및 성능 측정

<img width="900" alt="image" src="https://user-images.githubusercontent.com/49643709/194372398-64a2d4ae-9c9e-495e-a70d-adad94c0b002.png">
<img width="900" alt="image" src="https://user-images.githubusercontent.com/49643709/194372471-19f42632-25a4-4c01-b19f-f114a63d3bd0.png">
